browser/brave_browser_process_impl.cc

```diff
@@ -200,6 +202,10 @@ void BraveBrowserProcessImpl::StartBraveServices() {
   https_everywhere_service()->Start();
   resource_component();
 
+  if (base::FeatureList::IsEnabled(net::features::kBraveHttpsByDefault)) {
```

Comment:
this is definitely not the right place for this check. All this is doing is delaying the initialization if the feature is disabled. The check should go inside `HttpsUpgradeExceptionsService`